lib/ansible/modules/cloud/azure/azure_rm_sqlserver.py

```diff
@@ -0,0 +1,321 @@
+    returned: always
+    type: str
+    sample: /subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/sqlcrudtest-7398/providers/Microsoft.Sql/servers/sqlcrudtest-4645
+version:
+    description:
+        - The version of the server.
+    returned: always
+    type: str
+    sample: 12.0
+state:
+    description:
+        - The state of the server.
+    returned: always
+    type: str
+    sample: state
+fully_qualified_domain_name:
+    description:
+        - The fully qualified domain name of the server.
+    returned: always
+    type: str
+    sample: sqlcrudtest-4645.database.windows.net
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM SQL Server resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            admin_username=dict(
+                type='str',
+                required=False
+            ),
+            admin_password=dict(
+                type='str',
+                no_log=True,
+                required=False
+            ),
+            version=dict(
+                type='str',
+                required=False
+            ),
+            identity=dict(
+                type='str',
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+        self.parameters['identity'] = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
```

Comment:
What's the point of this whole block, `AzureRMModuleBase` should be setting `self.x` based on the module_arg_spec

Reply:
well, i have tried and it doesn't work as expected, so i have to leave it for now...

Reply:
Ok, will try and debug and find out what is happening, it should just work.

Reply:
Ok I see my mistake, most modules just have

```
for key in list(self.module_arg_spec.keys()) + ['tags']:
    setattr(self, key, kwargs[key])
```

Is there a reason why you have all these if statements as the above should be enough for you.

Reply:
yes, these attributes are sticked into objects padded to Azure SDK, so treated in different way